drivers/gpu/drm/rp1/rp1-dsi/rp1_dsi_dsi.c

```diff
@@ -1471,6 +1516,28 @@ void rp1dsi_dsi_send(struct rp1_dsi *dsi, u32 hdr, int len, const u8 *buf)
 		usleep_range(100, 150);
 	}
 
+	/*
+	 * Update global configuration flags for LP/HS and ACK options.
+	 * XXX It's not clear if having empty FIFOs (checked above and below) guarantees that
+	 * the last command has completed and been ACKed, or how closely these control registers
+	 * align with command/payload FIFO writes (as each is an independent clock-crossing)?
+	 */
+	if (use_lpm != dsi->using_lpm || req_ack != dsi->using_ack) {
+		val = DSI_READ(DSI_VID_MODE_CFG);
+		if (use_lpm)
+			val |= DSI_VID_MODE_LP_CMD_EN;
+		else
+			val &= ~DSI_VID_MODE_LP_CMD_EN;
+		DSI_WRITE(DSI_VID_MODE_CFG, val);
+		val = (use_lpm) ? DSI_CMD_MODE_ALL_LP : 0;
+		if (req_ack)
+			val |= DSI_CMD_MODE_ACK_RQST_EN;
+		DSI_WRITE(DSI_CMD_MODE_CFG, val);
+		(void)(0 + DSI_READ(DSI_CMD_MODE_CFG));
```

Comment:
Generally, I'd expect a to find a comment as to why a register read is there where the return value is not used.

Reply:
I actually do not know if it's needed or not.

I'm trying to look at the dsi_host RTL but have not yet tracked down how these (LP, ENA_ACK) signals are synchronized from the APB to the byteclock domain. I do hope they're not assumed to be constant. So far I suspect there's more risk the flags may be applied to a previous transmission rather than fail to be applied to the new one....

Reply:
I've removed the `0 +`.